### DISCUSSION TITLE
Persistent error in Snugdock: std::out_of_range: map::at:  key not found Got some signal... It is:6

### DISCUSSION THREAD

**Discussion by rexdog1** [RosettaCommons/rosetta#185] [Rosetta]
I am new to using command line software and am trying to perform flexible docking using Snugdock. I keep getting the following error when attempting to dock my (small molecule) antigen to the L and H chains of the antibody:

"AN INTERNAL ERROR HAS OCCURED. PLEASE SEE THE CONTENTS OF ROSETTA_CRASH.log FOR DETAILS.

libc++abi: terminating due to uncaught exception of type std::out_of_range: map::at:  key not found
Got some signal... It is:6
Signal 6 (SIGABRT) means that the process was aborted.  This usually means an internal Rosetta error caused by (often) bad inputs, (sometimes) developer error, or (rarely) hardware problems.
bash-3.2$ exit
exit

Script done on Tue Oct  1 12:33:46 2024"

[command_output.txt](https://github.com/user-attachments/files/17215990/command_output.txt)

I've attached the full output from the command line. From what I can gather, Rosetta is attempting to map a residue or atom and is unable to find it. I have meticulously double and triple checked, and as far as I can tell, all atoms are correctly named and labeled in the .pdb and match all of the atoms in the parameter file I generated for the ligand. The chains are labeled correctly as well, I think, but there could be formatting problems that I haven't been able to figure out yet. 

I also attached my flags file, which I've tried to assemble based on information available online, which can be somewhat inconsistent. 

[flags.txt](https://github.com/user-attachments/files/17216012/flags.txt)

Could there be problems with the flags I'm applying that is causing Rosetta to be unable to correctly assemble the fold tree and access all required atoms? Any help is greatly appreciated and I am happy to supply any other information/files that may be helpful in debugging.

**Comment by roccomoretti** [comment]
> attempting` to dock my (small molecule) antigen 

I'm guessing that's your problem. The SnugDock protocol attempts to set up a Foldtree based on the center-of-mass residues in each chain (heavy, light and antigen). However, when doing so it assumes that there's an "N", "C" and "CA" atom in that residue. If your antigen is a small molecules and it doesn't have one of those three atoms, then you'll get the error you saw.

The work-around you can try is to edit your ligand's parameter file (and the input PDB) such that your antigen residue has "N", "C" and "CA" atoms. (Note that which atoms in the ligand you choose to rename are arbitrary. It probably doesn't matter, so long as it's consistent. If you can pick atoms that are "backbone like", that would probably be best, but it's likely not necessary. -- in particular, you don't necessarily need the "N" atom to be a nitrogen. Rosetta should work with a carbon atom renamed to "N".) -- That's not to say you won't run into other issues trying to use SnugDock on a non-protein antigen, but that should at least fix the proximal issue.

**Reply by rexdog1** [reply]
Thanks so much for your reply - I was starting to suspect this was the problem because I was able to get Snugdock to run a few other protein/antigen pairs that were both proteins. Renaming the atoms does seem to have fixed that problem, although now I am getting a new error: 

core.kinematics.FoldTree: delete_jump_seqpos: deleting jump 1
core.kinematics.FoldTree: delete_jump_seqpos: renumbering jump 2 to 1

ERROR: Error in core::conformation::Conformation::residue(): The sequence position requested was 0.  Pose numbering starts at 1.
ERROR:: Exit from: src/core/conformation/Conformation.hh line: 513

Is this because my small molecule antigen is only a single "residue"?

**Reply by roccomoretti** [reply]
Possibly? If you have the ROSETTA_CRASH.log file, that would help track down where the issue may be coming from. (That is, why it's trying to look for position 0 in the first place.)
